File: app/src/main/java/com/photo/editor/picskills/photoeditorpro/activities/ShareSerpActivity.java
package com.photo.editor.picskills.photoeditorpro.activities;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.bumptech.glide.Glide;
import com.photo.editor.picskills.photoeditorpro.R;
import com.photo.editor.picskills.photoeditorpro.custom.MyBounceInterpolator;
import com.photo.editor.picskills.photoeditorpro.utils.Constants;
import com.photo.editor.picskills.photoeditorpro.utils.support.MyExceptionHandlerPix;
import com.photo.editor.picskills.photoeditorpro.utils.support.SupportedClass;

public class ShareSerpActivity extends ParentActivity {

    Uri mImgUri;
    ImageView mIvCreate;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_share);
        Thread.setDefaultUncaughtExceptionHandler(new MyExceptionHandlerPix(ShareSerpActivity.this));
        init();

        (findViewById(R.id.iv_home)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*FullScreenAdManager.fullScreenAdsCheckPref(ShareActivity.this, FullScreenAdManager.ALL_PREFS.ATTR_ON_HOME_SCREEN, new FullScreenAdManager.GetBackPointer() {
                    @Override
                    public void returnAction() {*/
                Intent finishIntent = new Intent(ShareSerpActivity.this, MainActivity.class);
                finishIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivity(finishIntent);
                overridePendingTransition(R.anim.enter, R.anim.exit);
              /*      }
                });*/
            }
        });
        (findViewById(R.id.iv_back)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
        (findViewById(R.id.iv_share)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                shareOnOthersApp();
            }
        });

    }

    public void init() {

        mIvCreate = (ImageView) findViewById(R.id.iv_create);

        mImgUri = Uri.parse(getIntent().getStringExtra(Constants.KEY_URI_IMAGE));

        Glide.with(ShareSerpActivity.this)
                .asBitmap()
                .load(mImgUri)
                .into(mIvCreate);

        final Animation myAnim = AnimationUtils.loadAnimation(this, R.anim.bounce);

        MyBounceInterpolator interpolator = new MyBounceInterpolator(0.2, 20);
        myAnim.setInterpolator(interpolator);

        mIvCreate.startAnimation(myAnim);
    }

    private void shareOnOthersApp() {
        SupportedClass.shareWithOther(ShareSerpActivity.this, getString(R.string.txt_app_share_info), mImgUri);
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onBackPressed() {
        this.finish();
        overridePendingTransition(R.anim.left_to_right, R.anim.right_to_left);
    }

}

File: app/src/main/java/com/photo/editor/picskills/photoeditorpro/activities/ShareSerpActivity.kt
package com.photo.editor.picskills.photoeditorpro.activities

import android.os.Bundle
import com.photo.editor.picskills.photoeditorpro.R
import com.photo.editor.picskills.photoeditorpro.utils.support.MyExceptionHandlerPix
import android.content.Intent
import android.net.Uri
import android.util.DisplayMetrics
import android.util.Log
import android.view.MenuItem
import android.view.View
import com.bumptech.glide.Glide
import android.view.animation.AnimationUtils
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdSize
import com.google.android.gms.ads.AdView
import com.photo.editor.picskills.photoeditorpro.custom.MyBounceInterpolator
import com.photo.editor.picskills.photoeditorpro.databinding.ActivityShareBinding
import com.photo.editor.picskills.photoeditorpro.utils.Constants
import com.photo.editor.picskills.photoeditorpro.utils.UtilsPak
import com.photo.editor.picskills.photoeditorpro.utils.support.SupportedClass
import java.lang.Exception

class ShareSerpActivity : ParentActivity() {

    private var mImgUri: Uri? = null

    private lateinit var mainBinding:ActivityShareBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainBinding = ActivityShareBinding.inflate(layoutInflater)
        setContentView(mainBinding.root)

        Thread.setDefaultUncaughtExceptionHandler(MyExceptionHandlerPix(this@ShareSerpActivity))

        bannerAds()

        init()

        mainBinding.ivHome.setOnClickListener {
            val finishIntent = Intent(this@ShareSerpActivity, MainActivity::class.java)
            finishIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
            startActivity(finishIntent)
            overridePendingTransition(R.anim.enter, R.anim.exit)
        }
        mainBinding.ivBack.setOnClickListener { onBackPressed() }
        mainBinding.ivShare.setOnClickListener { shareOnOthersApp() }
    }

    /**************************************Banner Ads *********************************/
    private fun bannerAds() {
        mainBinding.adLayout.visibility = View.VISIBLE
        mainBinding.adLayout.post { loadBanner() }
    }

    private fun loadBanner() {
        Log.e("myTag", "BannerAds")
        val adView = AdView(this)
        adView.adUnitId = UtilsPak.getBannerID()
        val adSize = adSize
        adView.setAdSize(adSize)
        mainBinding.adLayout.addView(adView)
        val adRequest = AdRequest.Builder().build()
        // Start loading the ad in the background.
        try {
            adView.loadAd(adRequest)
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }

    // Determine the screen width (less decorations) to use for the ad width.
    @Suppress("DEPRECATION")
    private val adSize: AdSize
        get() {
            // Determine the screen width (less decorations) to use for the ad width.
            val display = windowManager.defaultDisplay
            val outMetrics = DisplayMetrics()
            display.getMetrics(outMetrics)
            val density = outMetrics.density
            var adWidthPixels = mainBinding.adLayout.width.toFloat()
            // If the ad hasn't been laid out, default to the full screen width.
            if (adWidthPixels == 0f) {
                adWidthPixels = outMetrics.widthPixels.toFloat()
            }
            val adWidth = (adWidthPixels / density).toInt()
            return AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(this, adWidth)
        }

    private fun init() {

        mImgUri = Uri.parse(intent.getStringExtra(Constants.KEY_URI_IMAGE))

        Glide.with(this@ShareSerpActivity)
            .asBitmap()
            .load(mImgUri)
            .into(mainBinding.ivCreate)
        val myAnim = AnimationUtils.loadAnimation(this, R.anim.bounce)
        val interpolator = MyBounceInterpolator(0.2, 20.0)
        myAnim.interpolator = interpolator
        mainBinding.ivCreate.startAnimation(myAnim)
    }

    private fun shareOnOthersApp() {
        SupportedClass.shareWithOther(
            this@ShareSerpActivity,
            getString(R.string.txt_app_share_info),
            mImgUri
        )
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onBackPressed() {
        finish()
        overridePendingTransition(R.anim.left_to_right, R.anim.right_to_left)
    }
}